src/game_event.cpp

```diff
@@ -345,8 +345,13 @@ void Game_Event::CheckEventAutostart() {
 }
 
 void Game_Event::CheckEventCollision() {
-	if (trigger == RPG::EventPage::Trigger_collision) {
-		CheckEventTriggerTouch(GetX(),GetY());
+	if (trigger == RPG::EventPage::Trigger_collision
+			&& GetLayer() != RPG::EventPage::Layers_same
+			&& !Main_Data::game_player->IsMoveRouteOverwritten()
+			&& !Game_Map::GetInterpreter().IsRunning()
+			&& !Main_Data::game_player->InAirship()
+			&& Main_Data::game_player->IsInPosition(GetX(), GetY())) {
+		SetAsWaitingForegroundExecution(true, false);
```

Comment:
> Refactor trigger=collision and layer!=same logic 

where did all these conditions come from?

Reply:
Reverse engineered from my test cases. I cannot guarantee this is perfect. But the tests are pretty comprehensive. If you see anything missing from the tests, let me know

Reply:
I would also add that of everything in this PR, this was the most difficult to figure out.

Reply:
actually there isn't really anything new here compared to CheckEventTriggerTouch, the new "face hero" (first bool of SetAsForeground) is correct.